src/chunk.c

```diff
@@ -1980,57 +1956,42 @@ chunk_resurrect(const Hypertable *ht, const ChunkStub *stub)
  * context. The returned chunk needs to be copied into another memory context in
  * case it needs to live beyond the lifetime of the other data.
  */
-static Chunk *
-chunk_find(const Hypertable *ht, const Point *p, bool resurrect, bool lock_slices)
+static int
+chunk_point_find_chunk_id(const Hypertable *ht, const Point *p)
 {
-	ChunkStub *stub;
-	Chunk *chunk = NULL;
+	int chunk_id = 0;
 	ChunkScanCtx ctx;
 
 	/* The scan context will keep the state accumulated during the scan */
 	chunk_scan_ctx_init(&ctx, ht->space, p);
 
-	/* Abort the scan when the chunk is found */
-	ctx.early_abort = true;
-
-	/* Scan for the chunk matching the point */
-	chunk_point_scan(&ctx, p, lock_slices);
-
-	/* Find the stub that has N matching dimension constraints */
-	stub = chunk_scan_ctx_get_chunk_stub(&ctx);
-
-	chunk_scan_ctx_destroy(&ctx);
-
-	if (NULL != stub)
+	/* Scan all dimensions for slices enclosing the point */
+	List *all_slices = NIL;
+	for (int dimension_index = 0; dimension_index < ctx.space->num_dimensions; dimension_index++)
 	{
-		ChunkStubScanCtx stubctx = {
-			.stub = stub,
-		};
+		ts_dimension_slice_scan_list(ctx.space->dimensions[dimension_index].fd.id,
+									 p->coordinates[dimension_index],
+									 &all_slices);
+	}
 
-		/* Fill in the rest of the chunk's data from the chunk table, unless
-		 * the chunk is marked as dropped. */
-		chunk = chunk_create_from_stub(&stubctx);
+	/* Find constraints matching dimension slices. */
+	ListCell *lc;
+	foreach (lc, all_slices)
+	{
+		chunk_id =
+			ts_chunk_constraint_scan_by_dimension_slice_chunk_id((DimensionSlice *) lfirst(lc),
+																 &ctx,
+																 CurrentMemoryContext);
```

Comment:
Note that this will open and close the chunk_constraint table for every iteration in the loop. Previously, we found that this was often costly, so it is better to open the table outside the foreach loop and then only restarting the scan within the loop.